File: OOP PROJECT3/Source/employee.cpp
#include<iostream>
#include<cstdlib>
#include<sstream>
#include"planecomponent-employee.h"

using namespace std;

Employee::Employee(){ 
    
    string N_array[]{"Στυλιανός Ψαρά",
                     "Ανδρέας Γεωργίου",
                     "Μαρία Αντωνίου",
                     "Αντώνης Αντωνίου",
                     "Κυριακή Κυριάκου"};
    
    int n=rand()%5;

    name=N_array[n];
    
    cout<<"Employee just created!"<<endl;

}

Employee::~Employee(){ 
    
    cout<<"Employee to be destroyed!"<<endl;

}
 
string Employee::toString(){

    string str;
    string s;
    stringstream ss;

    ss<<ID;
    ss>>s;
    
    str.append("ID: ");
    str.append(s);
    str.append(" Employee Name: ");
    str.append(name);

    return str;
}

File: OOP PROJECTJAVA/src/askisi_java/employee.java
package askisi_java;

public abstract class employee {
	
	protected String name;
	
	public employee(){ 
	    
	    String[] N_array=new String[5];
	    N_array[0]="Antreas georgiou";
	    N_array[1]="Marios Kiriakou";
	    N_array[2]="Kiriaki Papadopoullou";
	    N_array[3]="Antonia Markou";
	    N_array[4]="Stilianos Psaras";
	    
	    int n=(int)(Math.random()*5);

	    name=N_array[n];
	    
	    System.out.println("Employee just created!");

	}
	
	public abstract void WorkOn(planecomponent x );
	
    public abstract void report();
}
